x/mongo/driver/session/client_session.go

```diff
@@ -174,17 +180,13 @@ func MaxClusterTime(ct1, ct2 bson.Raw) bson.Raw {
 
 // NewClientSession creates a Client.
 func NewClientSession(pool *Pool, clientID uuid.UUID, sessionType Type, opts ...*ClientOptions) (*Client, error) {
+	mergedOpts := mergeClientOptions(opts...)
+
 	c := &Client{
-		Consistent:  true, // set default
```

Comment:
Echoing the concern in https://github.com/mongodb/mongo-go-driver/pull/695#discussion_r664710947 I wonder if this resurfaced with [`c2a51f5` (#702)](https://github.com/mongodb/mongo-go-driver/pull/702/commits/c2a51f56c584d043cc321118e08e1c10a19cc3b2). The default value of `Consistent` on implicit sessions may have been inadvertently changed to `false`.

When an implicit session is created internally, `session.NewClientSession` is called directly (i.e. it bypasses the public facing `Client.StartSession`)

`mergeClientOptions` in `x/mongo/driver/session/options` does not set a default value of `Consistent` to true.

If that seems right, I'd suggest we add a unit test for `session.NewClientSession` with each combination of `Consistent` (unset, true, false) and `Snapshot` (unset, true, false) to ensure session options are processed as expected.

Reply:
On L208 below, `Consistent` is set to `!c.Snapshot`; that seems correct to me and means that the default value of `Consistent` on implicit sessions is still true when `Snapshot` is false.

I've added a new table test in `client_session_test` to ensure that `Consistent` and `Snapshot` are processed correctly in `NewClientSession`.

Reply:
🤦 I totally missed that. Thank you for adding the unit test though. That gives more confidence that we are creating implicit sessions as expected and will catch any regressions later.